cremalab/workout-tracker
# Feature user account login

## Reference
- Original [Asana Task](https://app.asana.com/0/622910146909793/622910146909802)
- Combined this  [Asana Task](https://app.asana.com/0/622910146909793/622910146909871) into this branch

## Overview

This PR adds a login screen and "authentication" - (currently can tell if a user is in the database but doesn't return correct auth status). Upon successful login, the user is redirected to the dashboard page.

### Notes

Other authentication strategies considered:
-hapi-auth-cookie. Will likely add this in later but was just looking for a way to redirect the user based on authentication and not get into cookies for now.
-Auth0. Free account didn't have an option for a local strategy

Lessons learned:
- Hapi doesn't work well with passport. There used to be library that connected the 2 (hapi-passport which was renamed to travelouge) but it was discontinued. 


## Testing Instructions

-npm install && cd client && npm install && cd ..
-npm start
-Click "Log In" link from home page
-Successful log in should redirect to blank dashboard
